apps/meteor/app/ui-message/client/messageBox/messageBoxActions.js

```diff
@@ -69,123 +69,12 @@ messageBox.actions.add('Add_files_from', 'Computer', {
 
 const canGetGeolocation = new ReactiveVar(false);
 
-const getGeolocationPermission = () =>
-	new Promise((resolve) => {
-		if (!navigator.permissions) {
-			resolve(true);
-		}
-		navigator.permissions.query({ name: 'geolocation' }).then(({ state }) => {
-			resolve(state);
-		});
-	});
-
-const getGeolocationPosition = () =>
-	new Promise((resolvePos) => {
-		navigator.geolocation.getCurrentPosition(
-			resolvePos,
-			() => {
-				resolvePos(false);
-			},
-			{
-				enableHighAccuracy: true,
-				maximumAge: 0,
-				timeout: 10000,
-			},
-		);
-	});
-
-const getCoordinates = async () => {
-	const status = await getGeolocationPermission();
-	if (status === 'prompt') {
-		let resolveModal;
-		const modalAnswer = new Promise((resolve) => {
-			resolveModal = resolve;
-		});
-		modal.open(
-			{
-				title: t('You_will_be_asked_for_permissions'),
-				confirmButtonText: t('Continue'),
-				showCancelButton: true,
-				closeOnConfirm: true,
-				closeOnCancel: true,
-			},
-			async (isConfirm) => {
-				if (!isConfirm) {
-					resolveModal(false);
-				}
-				const position = await getGeolocationPosition();
-				if (!position) {
-					const newStatus = getGeolocationPermission();
-					resolveModal(newStatus);
-				}
-				resolveModal(position);
-			},
-		);
-		const position = await modalAnswer;
-		return position;
-	}
-
-	if (status === 'denied') {
-		return status;
-	}
-
-	const position = await getGeolocationPosition();
-	return position;
-};
-
 messageBox.actions.add('Share', 'My_location', {
 	id: 'share-location',
 	icon: 'map-pin',
 	condition: () => canGetGeolocation.get(),
```

Comment:
this condition blocks the user to access the modal where can get the information about allowing the browser to share location

Reply:
As we discussed, we don't currently have a fallback for missing support on the position api, so we'll keep not showing the location button when not supported.